alkira/resource_alkira_fortinet.go

```diff
@@ -0,0 +1,247 @@
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+			},
+			"credential_id": {
+				Description: "ID of Fortinet Firewall credential managed by credential resource.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"cxp": {
+				Description: "The CXP where the service should be provisioned.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"instances": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Description: "An array containing properties for each Fortinet Firewall instance " +
+					"that needs to be deployed. The number of instances should be equal to " +
+					"max_instance_count.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Description: "The name of the Fortinet Firewall instance.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"serial_number": {
+							Description: "The serialNumber of the Fortinet Firewall instance. Required " +
+								"only when licenseType is BRING_YOUR_OWN.",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"credential_id": {
+							Description: "The id of the Fortinet Firewall instance credentials. " +
+								"Required only when licenseType is BRING_YOUR_OWN.",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"license_type": {
+				Description:  "Fortinet license type, either `BRING_YOUR_OWN` or `PAY_AS_YOU_GO`.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"BRING_YOUR_OWN", "PAY_AS_YOU_GO"}, false),
+			},
+			"management_server_ip": {
+				Description: "The IP addresses used to access the management server.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"management_server_segment": {
+				Description: "The segment used to access the management server. This segment " +
+					"must be present in the list of segments assigned to this Fortinet Firewall service.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"max_instance_count": {
+				Description: "The maximum number of Fortinet Firewall instances that should be " +
+					"deployed when auto-scale is enabled. Note that auto-scale is not supported " +
+					"with Fortinet at this time. max_instance_count must be greater than or " +
+					"equal to min_instance_count.",
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"min_instance_count": {
+				Description: "The minimum number of Fortinet Firewall instances that should be " +
+					" deployed at any point in time.",
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"name": {
+				Description: "Name of the Fortinet Firewall service.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"segment_names": {
```

Comment:
We tried to be consistent to use `segment_ids`. That' easier for cleaner dependencies...